tests/cli/test_cli.py

```diff
@@ -198,9 +199,9 @@ def test_init_default(self, parent_dir, cwd):
             result = CliRunner().invoke(cli, ["init"])
 
             assert result.exit_code == 0
-            assert (
-                captured.records[0].getMessage()
-                == f"Creating custom nodes folder in {cwd / 'src' / 'custom_nodes'}"
+            assert_msg_in_logs(
+                f"Creating custom nodes folder in {cwd / 'src' / 'custom_nodes'}",
+                captured.records,
```

Comment:
I'm not sure if these are working as intended. `assert_msg_in_logs()` returns a `bool` but does not actually asserts it. And there's no assertion done here on the returned value. 

Perhaps renaming the utility function to `msg_in_logs()` and then asserting it like `assert msg_in_logs()` will be less error prone?

Reply:
Have fixed the bugged helper method `assert_msg_in_logs`. Now it looks for a (partial) msg within the logs and `assert True/False` accordingly.

Also amended the DeclarativeLoader to change directory just prior to loading the pipeline, to minimise any potential side-effects on the runtime environment.